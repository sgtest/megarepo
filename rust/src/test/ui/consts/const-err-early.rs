#![deny(const_err)]

pub const A: i8 = -i8::MIN; //~ ERROR const_err
//~| WARN this was previously accepted by the compiler but is being phased out
pub const B: u8 = 200u8 + 200u8; //~ ERROR const_err
//~| WARN this was previously accepted by the compiler but is being phased out
pub const C: u8 = 200u8 * 4; //~ ERROR const_err
//~| WARN this was previously accepted by the compiler but is being phased out
pub const D: u8 = 42u8 - (42u8 + 1); //~ ERROR const_err
//~| WARN this was previously accepted by the compiler but is being phased out
pub const E: u8 = [5u8][1]; //~ ERROR const_err
//~| WARN this was previously accepted by the compiler but is being phased out

fn main() {
    let _a = A;
    let _b = B;
    let _c = C;
    let _d = D;
    let _e = E;
    let _e = [6u8][1];
}
