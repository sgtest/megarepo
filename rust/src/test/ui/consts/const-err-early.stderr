error: any use of this value will cause an error
  --> $DIR/const-err-early.rs:3:19
   |
LL | pub const A: i8 = -i8::MIN;
   | ------------------^^^^^^^^-
   |                   |
   |                   attempt to negate `i8::MIN`, which would overflow
   |
note: the lint level is defined here
  --> $DIR/const-err-early.rs:1:9
   |
LL | #![deny(const_err)]
   |         ^^^^^^^^^
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #71800 <https://github.com/rust-lang/rust/issues/71800>

error: any use of this value will cause an error
  --> $DIR/const-err-early.rs:5:19
   |
LL | pub const B: u8 = 200u8 + 200u8;
   | ------------------^^^^^^^^^^^^^-
   |                   |
   |                   attempt to compute `200_u8 + 200_u8`, which would overflow
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #71800 <https://github.com/rust-lang/rust/issues/71800>

error: any use of this value will cause an error
  --> $DIR/const-err-early.rs:7:19
   |
LL | pub const C: u8 = 200u8 * 4;
   | ------------------^^^^^^^^^-
   |                   |
   |                   attempt to compute `200_u8 * 4_u8`, which would overflow
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #71800 <https://github.com/rust-lang/rust/issues/71800>

error: any use of this value will cause an error
  --> $DIR/const-err-early.rs:9:19
   |
LL | pub const D: u8 = 42u8 - (42u8 + 1);
   | ------------------^^^^^^^^^^^^^^^^^-
   |                   |
   |                   attempt to compute `42_u8 - 43_u8`, which would overflow
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #71800 <https://github.com/rust-lang/rust/issues/71800>

error: any use of this value will cause an error
  --> $DIR/const-err-early.rs:11:19
   |
LL | pub const E: u8 = [5u8][1];
   | ------------------^^^^^^^^-
   |                   |
   |                   index out of bounds: the length is 1 but the index is 1
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #71800 <https://github.com/rust-lang/rust/issues/71800>

error: aborting due to 5 previous errors

